src/exec/mod.rs

```diff
@@ -315,26 +393,62 @@ mod tests {
                     ]),
                 ],
                 mode: ExecutionMode::OneByOne,
+                path_separator: None,
             }
         );
     }
 
     #[test]
     fn tokens_single_batch() {
         assert_eq!(
-            CommandTemplate::new_batch(&["echo", "{.}"]).unwrap(),
+            CommandTemplate::new_batch(&["echo", "{.}"], None).unwrap(),
             CommandTemplate {
                 args: vec![
                     ArgumentTemplate::Text("echo".into()),
                     ArgumentTemplate::Tokens(vec![Token::NoExt]),
                 ],
                 mode: ExecutionMode::Batch,
+                path_separator: None,
             }
         );
     }
 
     #[test]
     fn tokens_multiple_batch() {
-        assert!(CommandTemplate::new_batch(&["echo", "{.}", "{}"]).is_err());
+        assert!(CommandTemplate::new_batch(&["echo", "{.}", "{}"], None).is_err());
+    }
+
+    #[test]
+    fn generate_custom_path_separator() {
+        let arg = ArgumentTemplate::Tokens(vec![Token::Placeholder]);
+        macro_rules! check {
+            ($input:expr, $expected:expr) => {
+                assert_eq!(arg.generate($input, Some("#")), OsString::from($expected));
+            };
+        }
+
+        check!("foo", "foo");
+        check!("foo/bar", "foo#bar");
+        check!("/foo/bar/baz", "#foo#bar#baz");
+
+        // windows-specific absolute path prefixes
+        if cfg!(windows) {
```

Comment:
maybe move these to a separate `#[cfg(windows)]` test?

Reply:
sure, I'll do that